README.md

```diff
@@ -1,2 +1,199 @@
-# Fluent
-A brand new framework to build amazing android applications
+
+# Fluent 
+
+Fluent is a lightweight framework thats helps you to create an entirely Android Application following SOLID pattern with an unidirectional flow reactive or not.
+
+The main concepts in this architecture are : `State`, `View`, `Store`, `Job` and `Hub` 
+
+---
+>![fluent diagram](https://image.ibb.co/cAXPhH/Screen_Shot_2018_04_11_at_12_09_41.png)
+---
+
+## State
+
+We recommend to define one `State` for each one of your screens.
+
+It holds just `type() : StateType` and should be inherit to fit your needs. As the example above:
+
+```kotlin
+data class LoginState(val type: StateType = StateType.Initial) : State {
+  override fun type() = type
+  fun setType(newType: StateType) = copy(type = newType)
+}
+```
+
+### StateType
+
+We already have some default kind of states, you can create your own type inheriting this class.
+The default states are: `Initial`, `Loading`, `Success` and `Error`
+
+---
+
+>For now ahed we can have two kind of approaches with this framework: A non reactive and a total reactive way.
+
+## Non reactive way
+
+### View
+
+### Store
+
+### Job
+
+### Hub
+
+
+## Reactive way
+
+We work with Reactive Extension to follow the Reactive Manifesto
+
+### View
+
+That's the reference to your screen itself, you should create your kind of `View` that has specific `State` attached. 
+
+```kotlin
+interface LoginView : View<LoginState> {
+  fun doLoginClicks(): Observable<Unit>
+  fun activityResults(): Observable<Pair<Int, Intent>>
+}
+```
+
+> Stay aware to think in a way to declare all the user's actions and screen's actions related as `Observable`.
```

Comment:
This block can also be used in the first section as long as it has no Rx stuffs.

Reply:
In this case, what do you think is better: move to the first section or repeat that part in the non reactive way section?

Reply:
In fact, this View return `Observables` because the Rx approach.

Reply:
Yeah, you can keep this info about returning observables and in the first section you can mention that the user might extend the View to expose their events for the Hub.